### PR TITLE
Fix wheel install hermeticity.

### LINKED ISSUE #1520: Pex fails to generate hermetic wheel installs on Linux hosts when the install target path is >127 characters long
For example:

1. Hermetic:
```
$ rm -rf /tmp/pex_root && pex --pex-root /tmp/pex_root mypy_protobuf==2.4 --intransitive -o mypy_protobuf.pex
$ find /tmp/pex_root/installed_wheels/ -name "protoc-gen-mypy" | while read script; do echo "$script:" && head -3 $script; done 
/tmp/pex_root/installed_wheels/60974b0bc64909a12b2201d5d0093478d955e90a/mypy_protobuf-2.4-py3-none-any.whl/bin/protoc-gen-mypy:
#!python
# -*- coding: utf-8 -*-
import re
```

2. Not hermetic:
```
$ pex_root="/tmp/$(echo | sha512sum | cut -d' ' -f1)"
$ echo $pex_root
/tmp/be688838ca8686e5c90689bf2ab585cef1137c999b48c70b92f67a5c34dc15697b5d11c982ed6d71be1e1e7f7b4e0733884aa97c3f7a339a8ed03577cf74be09
$ echo $pex_root | wc -c
134
$ rm -rf $pex_root && pex --pex-root $pex_root mypy_protobuf==2.4 --intransitive -o mypy_protobuf.pex
/home/jsirois/.venv/pex/lib/python3.9/site-packages/pex/pex_bootstrapper.py:483: PEXWarning: The venv for /tmp/be688838ca8686e5c90689bf2ab585cef1137c999b48c70b92f67a5c34dc15697b5d11c982ed6d71be1e1e7f7b4e0733884aa97c3f7a339a8ed03577cf74be09/pip.pex/8781476b36a7a6b86d240e88ab26d74d7bc81315 at /tmp/be688838ca8686e5c90689bf2ab585cef1137c999b48c70b92f67a5c34dc15697b5d11c982ed6d71be1e1e7f7b4e0733884aa97c3f7a339a8ed03577cf74be09/venvs/6d965e35637dae05966e8439822c63a427c2a522/108a3ddc84230ab282ea6312e06cb68f51008ce5 has script shebangs of '#!/tmp/be688838ca8686e5c90689bf2ab585cef1137c999b48c70b92f67a5c34dc15697b5d11c982ed6d71be1e1e7f7b4e0733884aa97c3f7a339a8ed03577cf74be09/venvs/s/adaddbff/venv/bin/python3 -sE' with 173 characters. On some systems this may be too long and cause problems running the venv scripts. You may be able adjust PEX_ROOT from /tmp/be688838ca8686e5c90689bf2ab585cef1137c999b48c70b92f67a5c34dc15697b5d11c982ed6d71be1e1e7f7b4e0733884aa97c3f7a339a8ed03577cf74be09 to a shorter path as a work-around.
  pex_warnings.warn(
$ find $pex_root/installed_wheels/ -name "protoc-gen-mypy" | while read script; do echo "$script:" && head -3 $script; done 
/tmp/be688838ca8686e5c90689bf2ab585cef1137c999b48c70b92f67a5c34dc15697b5d11c982ed6d71be1e1e7f7b4e0733884aa97c3f7a339a8ed03577cf74be09/installed_wheels/60974b0bc64909a12b2201d5d0093478d955e90a/mypy_protobuf-2.4-py3-none-any.whl/bin/protoc-gen-mypy:
#!/bin/sh
'''exec' /tmp/be688838ca8686e5c90689bf2ab585cef1137c999b48c70b92f67a5c34dc15697b5d11c982ed6d71be1e1e7f7b4e0733884aa97c3f7a339a8ed03577cf74be09/venvs/s/adaddbff/venv/bin/python3 "$0" "$@"
' '''
```

### PR DESCRIPTION
**Pull request by jsirois**
Previously console scripts could be emitted in wheel install chroots
with non-hermetic launch headers when the PEX_ROOT was too long a path.

Fixes #1520

### CODE REVIEW

File: tests/integration/test_issue_1520.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    scripts = [
+        os.path.join(root, f)
+        for root, dirs, files in os.walk(os.path.join(pex_root, "installed_wheels"))
+        for f in files
+        if "protoc-gen-mypy" == f
+    ]
+    assert 1 == len(scripts)
+    with open(scripts[0]) as fp:
+        assert "#!python" == fp.readline().strip()
```
Comment (jsirois) [inline tests/integration/test_issue_1520.py:54]: Fails with `AssertionError: assert '#!python' == '#!/bin/sh'` pre-fix.

File: tests/integration/test_issue_1520.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        assert "# -*- coding: utf-8 -*-" == fp.readline().strip()
+
+    shutil.rmtree(pex_root)
+    # This should no-op since there is no proto sent on stdin and exit success.
+    subprocess.check_call(
```
Comment (jsirois) [inline tests/integration/test_issue_1520.py:59]: Fails with the following pre-fix:
```
tests/integration/test_issue_1520.py::test_hermetic_console_scripts /home/jsirois/.pex/venvs/6a485a634b4551f45f501552ee4a328b3fdff10b/9148e449944f6676b304f8f06462e05782fdb928/bin/protoc-gen-mypy: line 2: /tmp/pytest-of-jsirois/pytest-76/popen-gw0/test_hermetic_console_scripts0/_______________________________________________________________________________________________________________________________/pex_root/venvs/s/1ef662fb/venv/bin/python: No such file or directory
```

File: pex/pip.py [hunk 3]
```diff
@@ -1037,32 +1049,76 @@ def _fixup_record(
...
-                else:
+                elif (
+                    not first_non_shebang_line
+                    or cast(bytes, line).strip() == first_non_shebang_line
+                ):
                     # N.B.: These lines include the newline already.
                     buffer.write(cast(bytes, line))
+                    first_non_shebang_line = None
+                else:
+                    TRACER.log("Skipping garbage line: {}".format(line.strip()))
```
Comment (jsirois) [inline pex/pip.py:1121]: kill this leftover.

Comment (jsirois) [reply pex/pip.py]: Fixed.

**Review by Eric-Arellano — APPROVED** [review]
Thank you!

**Comment by jsirois** [comment]
Confirmed this fixes https://github.com/pantsbuild/pants/issues/13170

**Review by tdyas — APPROVED** [review]

**Pull request merged**